common/flatpak-dir.c

```diff
@@ -1970,18 +1970,29 @@ _flatpak_dir_ensure_repo (FlatpakDir   *self,
           GKeyFile *orig_config = NULL;
           g_autoptr(GKeyFile) new_config = NULL;
           g_autofree char *orig_min_free_space_percent = NULL;
+          g_autofree char *orig_min_free_space_size = NULL;
           g_autofree char *orig_add_remotes_config_dir = NULL;
 
           orig_config = ostree_repo_get_config (repo);
           orig_min_free_space_percent = g_key_file_get_value (orig_config, "core", "min-free-space-percent", NULL);
+          orig_min_free_space_size = g_key_file_get_value (orig_config, "core", "min-free-space-size", NULL);
           orig_add_remotes_config_dir = g_key_file_get_value (orig_config, "core", "add-remotes-config-dir", NULL);
 
-          if (orig_min_free_space_percent == NULL ||
+          if (orig_min_free_space_size == NULL ||
               orig_add_remotes_config_dir == NULL)
             new_config = ostree_repo_copy_config (repo);
 
-          if (orig_min_free_space_percent == NULL)
-            g_key_file_set_string (new_config, "core", "min-free-space-percent", "0");
+          /* Scrap previously written min-free-space-percent=0 and replace it with min-free-space-size */
+          if (orig_min_free_space_percent != NULL &&
+              g_ascii_strtoull (orig_min_free_space_percent, NULL, 0) == 0)
```

Comment:
`g_ascii_strtoull()` returns `0` on error, so this condition will also hold true if there’s an error parsing the setting.

You might want to use `g_ascii_string_to_unsigned()` instead, if flatpak depends on a recent enough version of GLib. It simplifies the error handling.